share/picongpu/tests/XrayScattering/include/picongpu/param/xrayScattering.param

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2020 Pawel Ordyna
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/plugins/xrayScattering/beam/Side.hpp"
+
+/* preprocessor struct generator */
+#include <pmacc/preprocessor/struct.hpp>
+
+#ifndef PARAM_ANGLES
+#   define PARAM_ANGLES 0
+#endif
+
+namespace picongpu
+{
+namespace plugins
+{
+namespace xrayScattering
+{
+namespace beam
+{
+    using namespace picongpu::plugins::xrayScattering::beam;
+    /* Choose from:
+     *  - ZSide
+     *  - YSide
+     *  - XSide
+     * - ZRSide
+     * - YRSide
+     * - XRSide
+     */
+    using ProbingSide = ZSide;
+
+    PMACC_STRUCT( RotationParam,
+        ( PMACC_C_VALUE ( float_X, yawAngle,
+#if PARAM_ANGLES == 1
+            20.0 / 180.0 * math::Pi< float_64 >::value
+#else
+        0.0
+#endif
+            ) )
+        ( PMACC_C_VALUE ( float_X, pitchAngle,
+#if PARAM_ANGLES == 1
+            42.0 / 180.0 * math::Pi< float_64 >::value
+#else
+            0.0
```

Comment:
Not a big deal, but for `float_X` arithmetic it would have been a little cleaner to write literal constants as `42.0_X` and use this type in templates like `math::Pi< float_X >`